dotCMS/src/main/java/com/dotcms/graphql/exception/CustomGraphQLException.java

```diff
@@ -0,0 +1,39 @@
+package com.dotcms.graphql.exception;
+
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.language.SourceLocation;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CustomGraphQLException extends RuntimeException implements GraphQLError {
+    private final int errorCode;
+
+    public CustomGraphQLException(int errorCode, String errorMessage) {
+        super(errorMessage);
+
+        this.errorCode = errorCode;
+    }
+
+    @Override
+    public Map<String, Object> getExtensions() {
+        Map<String, Object> customAttributes = new LinkedHashMap<>();
+
+        customAttributes.put("errorCode", this.errorCode);
+        customAttributes.put("errorMessage", this.getMessage());
+
+        return customAttributes;
```

Comment:
if we need to keep this.. of which I am not certain. 
The Pattern would be this: 
`return Map.of("errorCode",,this.errorCode,"errorMessage", this.getMessage());`
Try to use immutable class to return results as mush as possible

Reply:
![Screenshot 2024-08-09 at 3 36 22 PM](https://github.com/user-attachments/assets/bc2bb166-444f-47b1-8d8e-fe42f6cb7569)

The code returned by the endpoint is still 200, this code needs to be changed to match the permissions issue
We have exceptions to code mapping configured in `ReloadableServletContainer` Check it out, Perhaps all you need to do is use one of our own exceptions passing the right message and that's it.

Reply:
As far as I know, the HTTP status code returned is typically 200, regardless of whether the response contains data or errors. This is because the HTTP status code reflects the success of the HTTP request itself, not the success of the GraphQL query. While customizing status codes is possible, it’s not standard practice as it goes against GraphQL’s principle of being transport-agnostic.

If we still want the status code to match the specific error thrown by the server, we would need to implement middleware that inspects all error responses, categorizes each error based on its type, and sets the corresponding status code. However, this could become more complex if the response contains multiple errors. Given the scope of this issue, I believe this approach is beyond what is currently needed.

Reply:
you're right about this.